ui/src/app/pages/workspace/workspace-share.tsx

```diff
@@ -263,10 +271,22 @@ export const WorkspaceShare = fp.flow(withUserProfile())(
     }
 
     removeCollaborator(user: UserRole): void {
+      // remove from userRoles if it exists
```

Comment:
userRoles is what is displayed in the modal: simply remove the collab.

userRolesToChange is the changeset we use in the share request: we need to explicitly set [user, NOACCESS] to do this.  However!  we may have previously added the user to userRolesToChange, so we need to filter this out first.